lib/graphql_schema.rb

```diff
@@ -24,6 +24,10 @@ def types
     @types ||= @hash.fetch('types').map{ |type_hash| TypeDefinition.new(type_hash) }.sort_by(&:name)
   end
 
+  def type_from_name(type_name)
+    @types.find { |type| type.name == type_name }
```

Comment:
Yes but memoizing when `type_from_name` is used would mean you only have a performance increase if you're accessing the same types often.

We could create a look up table the first time `type_from_name` is used which would be the best of both worlds.